app/store/recipients.go

```diff
@@ -0,0 +1,235 @@
+	recipientUpdate    = "REPLACE INTO recipients (id, e164, uuid, username, email, is_blocked, profile_key, profile_key_credential, profile_given_name, profile_family_name, profile_joined_name, signal_profile_avatar, profile_sharing_enabled, last_profile_fetch, unidentified_access_mode, storage_service_id, storage_proto, capabilities, last_session_reset, about, about_emoji) VALUES (:id, :e164, :uuid, :username, :email, :is_blocked, :profile_key, :profile_key_credential, :profile_given_name, :profile_family_name, :profile_joined_name, :signal_profile_avatar, :profile_sharing_enabled, :last_profile_fetch, :unidentified_access_mode, :storage_service_id, :storage_proto, :capabilities, :last_session_reset, :about, :about_emoji)"
+	recipientGetById   = "SELECT * FROM recipients WHERE id = ?"
+	recipientGetByUUID = "SELECT * FROM recipients WHERE uuid = ? Limit 1"
+	recipientGetByE164 = "SELECT * FROM recipients WHERE e164 = ?"
+)
+
+type Recipient struct {
+	Id                     int64  `db:"id"`
+	E164                   string `db:"e164"`
+	UUID                   string `db:"uuid"`
+	Username               string `db:"username"`
+	Email                  string `db:"email"`
+	IsBlocked              bool   `db:"is_blocked"`
+	ProfileKey             []byte `db:"profile_key"`
+	ProfileKeyCredential   []byte `db:"profile_key_credential"`
+	ProfileGivenName       string `db:"profile_given_name"`
+	ProfileFamilyName      string `db:"profile_family_name"`
+	ProfileJoinedName      string `db:"profile_joined_name"`
+	SignalProfileAvatar    string `db:"signal_profile_avatar"`
+	ProfileSharingEnabled  bool   `db:"profile_sharing_enabled"`
+	LastProfileFetch       string `db:"last_profile_fetch"`
+	UnidentifiedAccessMode bool   `db:"unidentified_access_mode"`
+	StorageServiceId       []byte `db:"storage_service_id"`
+	StorageProto           []byte `db:"storage_proto"`
+	Capabilities           int    `db:"capabilities"`
+	LastSessionReset       string `db:"last_session_reset"`
+	About                  string `db:"about"`
+	AboutEmoji             string `db:"about_emoji"`
+}
+
+type Recipients struct {
+	Recipients []*Recipient `db:"recipients"`
+}
+
+var RecipientsModel = &Recipients{
+	Recipients: make([]*Recipient, 0),
+}
+
+// CreateRecipient creates a recipient
+func (r *Recipients) CreateRecipient(recipient *Recipient) (*Recipient, error) {
+	log.Debugln("[axolotl] Creating recipient", recipient.UUID)
+	storedRecipeit := r.GetRecipientByUUID(recipient.UUID)
+	if storedRecipeit != nil {
+		log.Debugln("[axolotl] CreateRecipient: Recipient already exists", recipient.UUID)
+		return storedRecipeit, nil
+	}
+	// get last inserted recipient id
+	var lastId int64
+	err := DS.Dbx.Get(&lastId, "SELECT id FROM recipients ORDER BY id DESC LIMIT 1;")
+	if err != nil {
+		lastId = 0
+	}
+	recipient.Id = lastId + 1
```

Comment:
Because it doesn't do it automatically. Don't know why, maybe this changed with the updated db driver.

Reply:
As far as I found out, it's because Id is initialized with the value 0.
You must not provide a value for id in the `recipientInsert` query, if you want that the database assigns a new id.
Regretfully, this means that `INSERT or REPLACE` can not leave the id assignment to the database.